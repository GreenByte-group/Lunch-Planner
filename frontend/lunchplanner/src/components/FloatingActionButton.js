import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import DoneIcon from '@material-ui/icons/Done'
import Button from '@material-ui/core/Button';

const styles = theme => ({
    whiteSymbol: {
        color: theme.palette.common.white
    },
    floatingButton: {
        position: 'absolute',
        float: 'right',
        bottom: '15px',
        marginLeft: 'calc(100% - 56px - 15px)',
    }
});


class FloatingActionButton extends React.Component {

    constructor(props) {
        super();
        this.state = {
            icon: props.icon,
        };
    }

    componentWillReceiveProps(newProps) {
        if(newProps.icon && newProps.icon !== this.state.icon)
            this.setState({
                icon: newProps.icon,
            })
    }

    render() {
        const { classes } = this.props;

        return (
            <Button variant="fab" color="secondary" className={classNames(classes.button, classes.whiteSymbol, classes.floatingButton)} onClick={this.props.onClick}>
                {(() => {switch (this.state.icon) {
                    case 'add':
                        return <AddIcon />;
                    case 'done':
                        return <DoneIcon />;
                    default:
                        return <AddIcon/>;

                }})()}
            </Button>
        )
    }

}

FloatingActionButton.propTypes = {
    classes: PropTypes.object.isRequired,
    theme: PropTypes.object.isRequired,
};

export default withStyles(styles, { withTheme: true })(FloatingActionButton);