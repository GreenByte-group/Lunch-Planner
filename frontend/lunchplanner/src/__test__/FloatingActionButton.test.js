import React from 'react';
import FloatingActionButton from '../components/FloatingActionButton';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <FloatingActionButton/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});