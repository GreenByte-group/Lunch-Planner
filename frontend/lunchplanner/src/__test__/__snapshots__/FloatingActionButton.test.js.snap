// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="MuiButtonBase-root-19 MuiButton-root-3 MuiButton-raised-9 MuiButton-fab-14 MuiButton-raisedSecondary-11 FloatingActionButton-whiteSymbol-1 FloatingActionButton-floatingButton-2"
  disabled={false}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-4"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-22"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-28"
  />
</button>
`;
